lib/auth/kakao.ts

```
@@ -0,0 +1,42 @@
+export function kakaoLogin(redirectUri?: string) {
+  if (!window?.Kakao || !window.Kakao.Auth) {
+    throw new AuthError('Kakao SDK is not loaded or initialized.');
+  }
+
+  const uri = redirectUri ?? new URL('/auth/callback', window.location.origin).toString();
+  window.Kakao.Auth.authorize({ redirectUri: uri, prompt: 'login' });
+}
+
+export async function getKakaoToken(code: string, redirectUri: string): Promise<string> {
+  const KAKAO_TOKEN_URL = 'https://kauth.kakao.com/oauth/token';
+  const REST_API_KEY = process.env.REST_API_KEY!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)